crates/red_knot_python_semantic/resources/mdtest/narrow/truthiness.md

```diff
@@ -220,6 +220,46 @@ else:
 reveal_type(y)  # revealed: A & ~AlwaysTruthy | A & ~AlwaysFalsy
 ```
 
+## Truthiness of classes
+
+```py
+class MetaAmbiguous(type):
+    def __bool__(self) -> bool: ...
+
+class MetaFalsy(type):
+    def __bool__(self) -> Literal[False]: ...
+
+class MetaTruthy(type):
+    def __bool__(self) -> Literal[True]: ...
+
+class MetaDeferred(type):
+    def __bool__(self) -> MetaAmbiguous: ...
+
+class AmbiguousClass(metaclass=MetaAmbiguous): ...
+class FalsyClass(metaclass=MetaFalsy): ...
+class TruthyClass(metaclass=MetaTruthy): ...
+class DeferredClass(metaclass=MetaDeferred): ...
+
+
+def _(
+    a: type[AmbiguousClass], t: type[TruthyClass], f: type[FalsyClass], d: type[DeferredClass],
+    ta: type[TruthyClass | AmbiguousClass],
+    af: type[AmbiguousClass] | type[FalsyClass],
+):
+    reveal_type(ta)  # revealed: type[TruthyClass] | type[AmbiguousClass]
+    if ta:
+        reveal_type(ta)  # revealed: type[TruthyClass] | type[AmbiguousClass] & ~AlwaysFalsy
+
+    reveal_type(af)  # revealed: type[AmbiguousClass] | type[FalsyClass]
+    if af:
+        reveal_type(af)  # revealed: type[AmbiguousClass] & ~AlwaysFalsy
+
+    # TODO: Emit a diagnostic (`d` is not valid in boolean context)
+    if d:
+        # TODO: Should be `Unknown`
+        reveal_type(d)  # revealed: type[DeferredClass] & ~AlwaysFalsy
```

Comment:
Something like this (plus adding `flag: bool` to the parameter list for above) should be sufficient to test the `ClassLiteral` case:
```suggestion
        reveal_type(d)  # revealed: type[DeferredClass] & ~AlwaysFalsy
        
        tf = TruthyClass if flag else FalsyClass
        reveal_type(tf)  # Literal[TruthyClass, FalsyClass]
        if tf:
            reveal_type(tf)  # Literal[TruthyClass]
        else:
            reveal_type(tf)  # Literal[FalsyClass]
```